### PR TITLE
Consolidate PillButton styles

### PR DESCRIPTION
**Pull request by cy255**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Add: PillButtonStyle.swift
- Renamed "filled" style to "onBrand" and "outline" style to "primary".
- Moved PillButton style and color implementations from PillButtonStyle enum to PillButton extension for clarity.
- Added background and title colors for highlighted state.
- Changed primary active background color in dark mode from a gray to primary color for window.

PillButton.swift
- Update button appearance when "isHighlighted" is updated
- Removed color constants
- Reordered file contents based on access level.
- Renamed usages of "filled" to "onBrand" and "outline" to "primary"

PillButtonBar.swift
- Renamed usages of "filled" to "onBrand" and "outline" to "primary"

PillButtonBarDemoController.swift
- Renamed usages of "filled" to "onBrand" and "outline" to "primary"

### Verification
Verified that the PillButtonBar adopts the expected title and background colors in both light and dark mode.

Before:

ConsolidatePillButtonColorsBefore

ConsolidatePillButtonColorsBefore_dark

After:

https://user-images.githubusercontent.com/8619051/104648884-ac2ab680-5668-11eb-93eb-ecf34c790c23.mov

https://user-images.githubusercontent.com/8619051/104653834-0e3aea00-5670-11eb-8431-e88fe99f3bc7.mov

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui-apple/pull/397)

### CODE REVIEW

File: ios/FluentUI/Pill Button Bar/PillButtonStyle.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+import UIKit
+
+// MARK: - PillButtonStyle
+
+@available(*, deprecated, renamed: "PillButtonStyle")
+public typealias MSPillButtonStyle = PillButtonStyle
+
+@objc(MSFPillButtonStyle)
+public enum PillButtonStyle: Int {
```
Comment (harrieshin) [inline ios/FluentUI/Pill Button Bar/PillButtonStyle.swift:14]: nit can we add what these onBrand and primary means?

**Review by harrieshin — APPROVED** [review]

**Review by anandrajeswaran — APPROVED** [review]

File: ios/FluentUI/Pill Button Bar/PillButton.swift [hunk 2]
```diff
@@ -203,19 +121,36 @@ open class PillButton: UIButton {
...
+                            ? PillButton.selectedHighlightedBackgroundColor(for: window, for: style)
+                            : PillButton.selectedBackgroundColor(for: window, for: style)
+                    }
+
+                    setTitleColor(customSelectedTextColor ?? PillButton.selectedTitleColor(for: window, for: style), for: .normal)
+                    setTitleColor(customSelectedTextColor ?? PillButton.selectedHighlightedTitleColor(for: window, for: style), for: .highlighted)
                 } else {
-                    backgroundColor = style.selectedDisabledBackgroundColor(for: window)
-                    setTitleColor(style.selectedDisabledTitleColor(for: window), for: .normal)
+                    backgroundColor = PillButton.selectedDisabledBackgroundColor(for: window, for: style)
```
Comment (rdeassis) [inline ios/FluentUI/Pill Button Bar/PillButton.swift:135]: Do we need to specify PillButton on these extension calls?
**PillButton**.selectedDisabledBackgroundColor(for: window, for: style)

File: ios/FluentUI/Pill Button Bar/PillButtonStyle.swift (hunk 1 above)
Comment (rdeassis) [inline ios/FluentUI/Pill Button Bar/PillButtonStyle.swift:14]: Not in the scope of this PR (and we also discussed about it offline), this change breaks the public API and the next release version ideally should reflect that (bumping at least the minor version).
@harrieshin we can discuss this further to make our release numbers more consistent with the rules of Semantic Versioning.

File: ios/FluentUI/Pill Button Bar/PillButton.swift (hunk 2 above)
Comment (cy255) [reply ios/FluentUI/Pill Button Bar/PillButton.swift:135]: **PillButton** is needed because the color extension calls are made static.

**Pull request merged**